test/e2e/apps/statefulset.go

```diff
@@ -609,21 +642,28 @@ var _ = SIGDescribe("StatefulSet", func() {
 			e2estatefulset.WaitForRunningAndReady(c, 3, ss)
 
 			ginkgo.By("Verifying that stateful set " + ssName + " was scaled up in order")
-			expectedOrder := []string{ssName + "-0", ssName + "-1", ssName + "-2"}
-			ctx, cancel := watchtools.ContextWithOptionalTimeout(context.Background(), statefulSetTimeout)
-			defer cancel()
-			_, err = watchtools.Until(ctx, pl.ResourceVersion, w, func(event watch.Event) (bool, error) {
-				if event.Type != watch.Added {
-					return false, nil
-				}
-				pod := event.Object.(*v1.Pod)
-				if pod.Name == expectedOrder[0] {
-					expectedOrder = expectedOrder[1:]
-				}
-				return len(expectedOrder) == 0, nil
-
-			})
-			framework.ExpectNoError(err)
+			wg.Wait()
+			framework.ExpectNoError(orderErr)
+
+			// Verify that statuful set will be scaled down in order.
```

Comment:
move this block after `Scale down will halt with unhealthy stateful pod` and revert to watchtools.Until starting at pl.ResourceVersion for this PR

Reply:
done